### ISSUE TITLE
Required headers to connect RUM and Traces are not being added when using Flutter web

### ISSUE THREAD

**Issue by Gummiees** [DataDog/dd-sdk-flutter#554]
**Describe what happened**

I have set up a Datadog RUM **Flutter web only** application successfully, but when connecting RUM and traces [following the docs](https://docs.datadoghq.com/real_user_monitoring/mobile_and_tv_monitoring/setup/flutter/#setup), it does not set the headers.

When opening the DevTools and checking the console, Datadog has printed multiple warnings (one per first party host), saying:

```
datadog-rum-slim.js:1 Datadog Browser SDK: Allowed Tracing Urls parameters should be a string, RegExp, function, or an object. Ignoring parameter 
```

<details>
<summary>Full Chrome printed log</summary>

```
datadog-rum-slim.js:1 Datadog Browser SDK: Allowed Tracing Urls parameters should be a string, RegExp, function, or an object. Ignoring parameter Object
(anonymous) @ datadog-rum-slim.js:1
(anonymous) @ datadog-rum-slim.js:1
wn @ datadog-rum-slim.js:1
(anonymous) @ datadog-rum-slim.js:1
l @ datadog-rum-slim.js:1
(anonymous) @ datadog-rum-slim.js:1
initialize @ ddrum_web.dart:27
initialize @ datadog_flutter_plugin_web.dart:77
runBody @ async_patch.dart:84
_async @ async_patch.dart:127
initialize @ datadog_flutter_plugin_web.dart:55
initialize @ datadog_flutter_plugin.dart:168
(anonymous) @ async_patch.dart:45
runUnary @ zone.dart:1661
handleValue @ future_impl.dart:162
handleValueCallback @ future_impl.dart:846
_propagateToListeners @ future_impl.dart:875
[_completeWithValue] @ future_impl.dart:647
(anonymous) @ future_impl.dart:721
_microtaskLoop @ schedule_microtask.dart:40
_startMicrotaskLoop @ schedule_microtask.dart:49
(anonymous) @ async_patch.dart:181
Promise.then (async)
_scheduleImmediateWithPromise @ async_patch.dart:179
_scheduleImmediate @ async_patch.dart:151
_scheduleAsyncCallback @ schedule_microtask.dart:69
_rootScheduleMicrotask @ zone.dart:1466
scheduleMicrotask @ zone.dart:1681
[_asyncCompleteWithValue] @ future_impl.dart:720
[_asyncComplete] @ future_impl.dart:680
_Future.immediate @ future_impl.dart:287
value @ future.dart:352
(anonymous) @ internal.dart:160
run @ zone.dart:1655
get nullFuture @ internal.dart:160
desc.get @ operations.dart:1430
cancel @ html_dart2js.dart:37252
_cancelAndValue @ stream_pipe.dart:57
(anonymous) @ stream.dart:1589
_checkAndCall @ operations.dart:574
dcall @ operations.dart:579
(anonymous) @ html_dart2js.dart:37268
load (async)
[_addEventListener] @ html_dart2js.dart:15856
[dartx.addEventListener] @ html_dart2js.dart:15844
[_tryResume] @ html_dart2js.dart:37298
onData @ html_dart2js.dart:37269
get first @ stream.dart:1588
send @ browser_client.dart:66
(anonymous) @ async_patch.dart:45
runUnary @ zone.dart:1661
handleValue @ future_impl.dart:162
handleValueCallback @ future_impl.dart:846
_propagateToListeners @ future_impl.dart:875
[_completeWithValue] @ future_impl.dart:647
(anonymous) @ future_impl.dart:721
_microtaskLoop @ schedule_microtask.dart:40
_startMicrotaskLoop @ schedule_microtask.dart:49
(anonymous) @ async_patch.dart:181
Promise.then (async)
_scheduleImmediateWithPromise @ async_patch.dart:179
_scheduleImmediate @ async_patch.dart:151
_scheduleAsyncCallback @ schedule_microtask.dart:69
_rootScheduleMicrotask @ zone.dart:1466
scheduleMicrotask @ zone.dart:1681
[_asyncCompleteWithValue] @ future_impl.dart:720
[_asyncComplete] @ future_impl.dart:680
runBody @ async_patch.dart:112
_async @ async_patch.dart:127
handleFrameworkMessage @ plugin_registry.dart:116
[_sendPlatformMessage] @ platform_dispatcher.dart:629
sendPlatformMessage @ platform_dispatcher.dart:354
[_setAppLifecycleState] @ platform_dispatcher.dart:964
_engine.EnginePlatformDispatcher.new @ platform_dispatcher.dart:70
get _instance @ platform_dispatcher.dart:75
desc.get @ operations.dart:1430
get instance @ platform_dispatcher.dart:74
get window @ window.dart:376
desc.get @ operations.dart:1430
_engine.FlutterViewEmbedder.new @ embedder.dart:42
ensureFlutterViewEmbedderInitialized @ embedder.dart:332
initializeEngineUi @ initialization.dart:223
runBody @ async_patch.dart:84
_async @ async_patch.dart:127
initializeEngineUi @ initialization.dart:207
(anonymous) @ initialization.dart:39
runBody @ async_patch.dart:84
_async @ async_patch.dart:127
(anonymous) @ initialization.dart:35
_checkAndCall @ operations.dart:574
callMethod @ operations.dart:644
dsend @ operations.dart:648
(anonymous) @ app_bootstrap.dart:56
runBody @ async_patch.dart:84
_async @ async_patch.dart:127
(anonymous) @ app_bootstrap.dart:55
(anonymous) @ js_loader.dart:74
_checkAndCall @ operations.dart:574
dcall @ operations.dart:579
ret @ js_allow_interop_patch.dart:17
(anonymous) @ VM10:10
Promise.then (async)
onEntrypointLoaded @ VM10:9
didCreateEngineInitializer @ flutter.js:272
bootstrapEngine @ initialization.dart:54
runBody @ async_patch.dart:84
_async @ async_patch.dart:127
bootstrapEngine @ initialization.dart:27
main @ web_entrypoint.dart:19
runBody @ async_patch.dart:84
_async @ async_patch.dart:127
main @ web_entrypoint.dart:18
(anonymous) @ main_module.bootstrap.js:23
window.$dartRunMain @ main_module.bootstrap.js:22
(anonymous) @ unknown
runMain @ client.js:8979
(anonymous) @ client.js:24972
(anonymous) @ client.js:3589
call$2 @ client.js:11698
_asyncStartSync @ client.js:3553
$call$body$main__closure @ client.js:24985
call$1 @ client.js:24911
_rootRunUnary @ client.js:3948
runUnary$2$2 @ client.js:13095
runUnaryGuarded$1$2 @ client.js:13042
_sendData$1 @ client.js:12635
_add$1 @ client.js:12581
_add$1 @ client.js:12916
_handleData$2 @ client.js:12979
_handleData$1 @ client.js:12942
(anonymous) @ client.js:1348
_rootRunUnary @ client.js:3948
runUnary$2$2 @ client.js:13095
runUnaryGuarded$1$2 @ client.js:13042
_sendData$1 @ client.js:12635
_add$1 @ client.js:12581
_sendData$1 @ client.js:12427
add$1 @ client.js:12306
(anonymous) @ client.js:1406
_rootRunUnary @ client.js:3948
runUnary$2$2 @ client.js:13095
runUnaryGuarded$1$2 @ client.js:13042
_sendData$1 @ client.js:12635
_add$1 @ client.js:12581
_sendData$1 @ client.js:12427
add$1 @ client.js:12306
add$1 @ client.js:24411
call$1 @ client.js:24701
call$1 @ client.js:18104
_rootRunUnary @ client.js:3952
runUnary$2$2 @ client.js:13095
runUnaryGuarded$1$2 @ client.js:13042
call$1 @ client.js:13232
invokeClosure @ client.js:1238
(anonymous) @ client.js:1257
```
</details>

I've tried using `datadog_flutter_plugin` v2.0.0, v2.1.0, v2.1.1 and v2.2.0 with no success. I've also tried using the Browser SDK v4 instead of v5 (since the README mentions both).

**Steps to reproduce the issue:**

1. Setup a Flutter project with the desired Datadog configuration including RUM ([following this Datadog doc](https://docs.datadoghq.com/real_user_monitoring/mobile_and_tv_monitoring/setup/flutter/#setup))
2. Connect RUM and traces ([following this other Datadog doc](https://docs.datadoghq.com/real_user_monitoring/connect_rum_and_traces/?tab=w3ctracecontext))

Now having a configuration that looks like this:

```dart
DatadogConfiguration(
  clientToken: '<CLIENT_TOKEN>',
  env: '<ENV_NAME>',
  site: DatadogSite.eu1,
  nativeCrashReportEnabled: true,
  loggingConfiguration: DatadogLoggingConfiguration(),
  firstPartyHosts: ['example.com'],
  service: '<SERVICE_NAME>',
  rumConfiguration: DatadogRumConfiguration(applicationId: '<RUM_APPLICATION_ID>');
)..enableHttpTracking();
```

**Describe what you expected:**

To get the expected headers on the request to the set `firstPartyHosts`.

**Additional context**
 - Dart/Flutter version: 3.16.9
 - Android/iOS OS version: Irrelevant, using only web. Chrome 21.0.6167.85
 - Device Model: MacBook Pro (M1 Pro) version 14.2.1 (23C71)
 - Datadog SDK version: 
   - `datadog_flutter_plugin`: 2.2.0
   - `datadog_tracking_http_client`: 2.1.0
   - `datadog-logs.js`: v5
   - `datadog-rum-slim`: v5

**Comment by fuzzybinary** [comment]
Hi @Gummiees ,

Thanks for reporting the issue. It looks like the Datadog Browser SDK has started checking if our `firstPartyHosts` use a `RegExp`, which they do, but the Dart `RegExp` type and the Javascript `RegExp` type are not the same.

I'll look into this and see if I can provide a patch early next week.

**Comment by fuzzybinary** [comment]
This should now be available in 2.3.0.

Let me know if you continue to see any issues.
